src/objectDetection/index.ts

```diff
@@ -153,24 +156,24 @@ export function useObjectDetection(
       .then((handle) => {
         console.log("useObjectDetection", runningMode, model, handle);
         processor.value = { detectorHandle: handle, plugin };
+        setDetectorHandle(handle);
       });
     return () => {
       console.log("useObjectDetection.useEffect.unsub", "releaseDetector");
-      if (processor.value?.detectorHandle !== undefined) {
-        getObjectDetectionModule().releaseDetector(
-          processor.value.detectorHandle
-        );
+      if (detectorHandle !== undefined) {
+        getObjectDetectionModule().releaseDetector(detectorHandle);
       }
     };
+    // The linter wants `processor` and `detectorHandle` in the dependency array, but these are
+    //  _set_ by this useEffect. Making this reactive based on these values would cause an infinite loop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
disabling react-hooks/exhaustive-deps is almost never a good idea. Yes, the infinite loop problem is a very tricky one but by removing detectorHandle from the deps you have introduced a subtle bug where the correct detectorhandle will not get released when this component gets shut down. 

It is for this very reason that I did not use a `useState` for detectorHandle. Probably best to discuss this in person. hooks can be tricky

Reply:
Okay - I am definitely not happy disabling linter rules without fully understanding things.  The problem I have here is that  according to my observation and [this documentation](https://docs.swmansion.com/react-native-reanimated/docs/core/useSharedValue/#remarks) `useShareValue` is not reactive on the JS side of things.  So, when `createDetector.then` sets the shared value, it doesn't cause a re-evaluate, so we never have the right handle.  At least that's what I was seeing on my machine, did the callbacks actually work for you?

In any case, I'll dig into alternate ways of getting the reactivity working correctly without disabling the lining rule if I don't hear anything from you.

Reply:
er, the act of setting the sharedvalue does not stimulate... is what I meant to say

Reply:
> You are correct. the act of getting the callback does not stimulate a re-render.
> 
> The re-render will be stimulated when we have some state that is being set in the onResults callback that we pass to `useObjectDetection`

Agreed. The fundamental problem here is an asynchronous issue.  The initial call of the `useLayoutEffect` happens when `processor.value` is its initial value of `undefined`, so it does nothing.  Then `createDetector` completes and sets `processor.value`.  But since that is not reactive, the `useLayoutEffect` never gets called again, our `detectorMap` is never set up, and we never propagate the events to the container.

I tried a number of variations, like getting rid of `useLayoutEffect` and just setting up the map in the callback from `getDetector` in `useEffect`.  That seems like it should work, even though that makes the useEffect depend on the callbacks and it means building a new detector if the consumer changes the callbacks.  But it just crashes invokes useEffect twice and silently fails.

No urgency on this - I have a working (if unacceptable long-term) version of the code to continue playing with if I decide to continue on filling out the sample in the next few days.

Reply:
I see. Interesting. I will look at this later today.